commons-core/src/main/scala/com/avsystem/commons/misc/Timestamp.scala

```diff
@@ -0,0 +1,48 @@
+package com.avsystem.commons
+package misc
+
+import com.avsystem.commons.serialization.{GenCodec, GenKeyCodec, IsoInstant}
+
+import scala.concurrent.duration.{FiniteDuration, TimeUnit}
+
+/**
+  * Millisecond-precision, general purpose, cross compiled timestamp representation.
```

Comment:
Yes, I am aware of the precision loss but I think it's still worth the cost of having to remember about it.

Reply:
I mean that this new representation has neat conversions to all the different "times" one can have to use. Its loss of precision disables us from using it as an intermediate format.

Reply:
Precision loss happens only at converting `Instant` to `Timestamp` - all the other conversions are fine. I can change the name of `toTimestamp` extension method on `Instant` to `truncateToTimestamp` so that it's clear that it loses precision. WDYT?

Reply:
That sgtm